src/ai/defaultai.cc

```diff
@@ -2159,6 +2174,34 @@ void DefaultAI::update_buildable_field(BuildableField& field) {
 		field.military_score_ += part;
 	}
 
+	// Using F-neurons to slightly modify calculated score
```

Comment:
what is the purpose of the following code

Reply:
F-neurons takes up to 5 bools and returns one bool. So this is just add possibility for AI to do some more magic. Will require training though. Yes, it is kind of black magic